core/src/main/scala/sbt/librarymanagement/LibraryManagementInterface.scala

```diff
@@ -0,0 +1,84 @@
+package sbt.librarymanagement
+
+import java.io.File
+import sbt.util.Logger
+
+/**
+ * Interface for library management intended for library management engine authors.
+ */
+abstract class LibraryManagementInterface {
```

Comment:
retrieve is a specific feature so a generic sounding name would be `DependencyResolutionInterface`.